my-vault/internal/handlers/handlers.go

```diff
@@ -0,0 +1,205 @@
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		logger.Error("Unexpected internal error", "error", err)
+		return
+	}
+	// build new secretObject
+	temp, err := NewCreateSecret(secretRequestObject)
+	if err != nil {
+		secretErr := errors.New("unexpected internal error")
+		http.Error(w, secretErr.Error(), http.StatusInternalServerError)
+		logger.Error("Unexpected internal error", "error", err)
+		return
+	}
+	//err = db.PutNewSecret(temp)
+	err = h.DB.PutNewSecret(temp)
+
+	if err != nil {
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		logger.Error("Unexpected internal error", "error", err)
+		return
+	}
+
+	res := CreateSecretResponse{
+		Id:      temp.Id,
+		Name:    temp.Name,
+		Message: fmt.Sprintf("Successfully created secret with secretId: %s and name: %s", temp.Id, temp.Name),
+	}
+	//logger.Info("Successfully saved secret", "secret_id", temp.Id)
+	logger.Info("Successfully saved secret", "secret_id", temp.Id, "name", temp.Name)
+	w.Header().Set("Content-Type", "application/json")
+	encoder := json.NewEncoder(w)
+	err = encoder.Encode(res)
+	if err != nil {
+
+		userErr := errors.New("unexpected internal error")
+		http.Error(w, userErr.Error(), http.StatusInternalServerError)
+		logger.Error("Unexpected internal error", "error", err)
+		return
+	}
+}
+
+func (h *SecretHandler) HandleGetSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretID := r.PathValue("secret_id")
+	secretEntry, err := h.DB.GetScretsById(secretID)
+	logger.Info("secret from Db", "secret", secretEntry)
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			http.Error(w, fmt.Sprintf("Secret with ID %s not found", secretID), http.StatusNotFound)
+			return
+		}
+		logger.Error("Unexpected internal error", "error", err)
+		http.Error(w, "Unexpected internal error", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	getSecretResponse := NewSecretResponseObject(secretEntry)
+	json.NewEncoder(w).Encode(getSecretResponse)
+	fmt.Printf("Retrieved secret----------------------------------------- %s", getSecretResponse)
+}
+
+func (h *SecretHandler) HandleDeleteSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretId := r.PathValue("secret_id")
+	if len(secretId) == 0 {
+		http.Error(w, "empty secret ID", http.StatusBadRequest)
+	}
+	secret, err := h.DB.DeleteSecretByID(secretId)
+	if err == storage.ErrNotFound {
+		http.Error(w, fmt.Sprintf("Secret with ID %s not found", secret), http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		http.Error(w, "Unexpected internal error", http.StatusInternalServerError)
+		logger.Error("Failed to delete secret", "error", err, "secret_id", secretId)
+		return
+	}
```

Comment:
It's nicer to scope all error handling to a single scope
```suggestion
	if err != nil {
		if errors.Is(err, storage.ErrNotFound) {
			http.Error(w, fmt.Sprintf("Secret with ID %s not found", secret), http.StatusNotFound)
			return
		}
		http.Error(w, "Unexpected internal error", http.StatusInternalServerError)
		logger.Error("Failed to delete secret", "error", err, "secret_id", secretId)
		return
	}
```